components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/logout/idpinit/processor/FederatedIdpInitLogoutProcessor.java

```diff
@@ -0,0 +1,632 @@
+import org.wso2.carbon.identity.oauth2.util.JWTSignatureValidationUtils;
+import org.wso2.carbon.idp.mgt.IdentityProviderManagementException;
+import org.wso2.carbon.idp.mgt.IdentityProviderManager;
+
+import java.text.ParseException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.regex.Matcher;
+
+import javax.servlet.http.HttpServletResponse;
+
+import static org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants.AUTHENTICATOR_NAME;
+import static org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants.BackchannelLogout.DEFAULT_IAT_VALIDITY_PERIOD;
+import static org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants.OIDC_BACKCHANNEL_LOGOUT_ENDPOINT_URL_PATTERN;
+import static org.wso2.carbon.identity.application.authenticator.oidc.util.OIDCErrorConstants.ErrorMessages;
+
+/**
+ * Processes the OIDC federated idp initiated logout requests.
+ */
+public class FederatedIdpInitLogoutProcessor extends IdentityProcessor {
+
+    private static final Log log = LogFactory.getLog(FederatedIdpInitLogoutProcessor.class);
+
+    @Override
+    public IdentityResponse.IdentityResponseBuilder process(IdentityRequest identityRequest) throws FrameworkException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Started processing OIDC federated IDP initiated logout request.");
+        }
+        return handleOIDCFederatedLogoutRequest(identityRequest);
+    }
+
+    /**
+     * Handles the logout request according to OIDC Back-channel logout specification.
+     *
+     * @param logoutRequest
+     * @return IdentityResponse.IdentityResponseBuilder.
+     * @throws LogoutClientException Exception occurred due to error in the logout request.
+     * @throws LogoutServerException Exception occurred from IS.
+     */
+    protected IdentityResponse.IdentityResponseBuilder handleOIDCFederatedLogoutRequest(
+            IdentityRequest logoutRequest) throws LogoutClientException, LogoutServerException {
+
+        try {
+            String logoutToken = logoutRequest.getParameter(OIDCAuthenticatorConstants.LOGOUT_TOKEN);
+            if (StringUtils.isBlank(logoutToken)) {
+                throw new LogoutClientException(ErrorMessages.LOGOUT_TOKEN_EMPTY_OR_NULL.getCode(),
+                        ErrorMessages.LOGOUT_TOKEN_EMPTY_OR_NULL.getMessage());
+            }
+
+            if (log.isDebugEnabled()) {
+                log.debug("Handling the OIDC federated IdP Initiated logout request for the obtained logout token: " +
+                        logoutToken);
+            }
+            // Get the claim set from the logout token.
+            SignedJWT signedJWT = SignedJWT.parse(logoutToken);
+            JWTClaimsSet claimsSet = signedJWT.getJWTClaimsSet();
+            // Check for the iss value in claim set.
+            isIssuerClaimExists(claimsSet);
```

Comment:
isIssuerClaimExists name usually implies there is a boolean return
Can we think of a better name?

validateIssuer(claimsSet) or similar